### PR TITLE
Improve exception message when store not found for cycle, assert TableDirectoryListing is not used in readOnly queues

### PR DESCRIPTION
**Pull request by nicktindall**

### CODE REVIEW

**Review by JerryShea — APPROVED** [review]

File: src/main/java/net/openhft/chronicle/queue/impl/single/StoreTailer.java [hunk 1]
```diff
@@ -749,7 +749,7 @@ private long approximateLastCycle2(int lastCycle) throws StreamCorruptedExceptio
                 lastCycle, queue.epoch(), false, this.store);
         this.setCycle(lastCycle);
         if (wireStore == null)
-            throw new IllegalStateException("Store not found for cycle " + Long.toHexString(lastCycle) + ". Probably the files were removed?");
+            throw new IllegalStateException("Store not found for cycle " + Long.toHexString(lastCycle) + ". Probably the files were removed? lastCycle=" + lastCycle + ", queue=" + queue.fileAbsolutePath());
```
Comment (JerryShea) [inline src/main/java/net/openhft/chronicle/queue/impl/single/StoreTailer.java]: we already output lastCycle

Comment (nicktindall) [reply src/main/java/net/openhft/chronicle/queue/impl/single/StoreTailer.java]: hehe good point...

Comment (nicktindall) [reply src/main/java/net/openhft/chronicle/queue/impl/single/StoreTailer.java]: I copied that from below

File: src/main/java/net/openhft/chronicle/queue/impl/single/TableDirectoryListing.java [hunk 2]
```diff
@@ -31,11 +33,13 @@
             final @NotNull TableStore<?> tableStore,
             final Path queuePath,
             final ToIntFunction<File> fileToCycleFunction,
-            final boolean readOnly) {
+            final boolean readOnly,
```
Comment (JerryShea) [inline src/main/java/net/openhft/chronicle/queue/impl/single/TableDirectoryListing.java]: you could drop readOnly as an argument and assert !tableStore.readOnly()

Comment (nicktindall) [reply src/main/java/net/openhft/chronicle/queue/impl/single/TableDirectoryListing.java]: Yep, love it

**Pull request merged**